cpp/csp/cppnodes/statsimpl.h

```diff
@@ -1214,52 +1221,50 @@ class Quantile
                     break;
             }
         #else
-            auto it = m_tree.begin();
-            std::advance( it, ft );
-            switch ( m_interpolation )
+            switch (m_interpolation)
             {
-                case LINEAR:
-                    if( ft == target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        double lower = *it;
-                        double higher = *++it;
-                        qtl = ( 1 - target + ft ) * lower + ( 1 - ct + target ) * higher;
-                    }
-                    break;
-                case LOWER:
-                    qtl = *it;
-                    break;
-                case HIGHER:
-                    qtl = ( ft == ct ? *it : *++it );
-                    break;
-                case MIDPOINT:
-                    if( ft == target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        double lower = *it;
-                        double higher = *++it;
-                        qtl = ( higher+lower ) / 2;
-                    }
-                    break;
-                case NEAREST:
-                    if( target - ft <= ct - target )
-                    {
-                        qtl = *it;
-                    }
-                    else
-                    {
-                        qtl = *++it;
-                    }
-                    break;
-                default:
-                    break;
+            case LINEAR:
+                if (ft == target)
+                {
+                    qtl = *m_tree.get<0>().nth(ft);
+                }
+                else
+                {
+                    double lower = *m_tree.get<0>().nth(ft);
+                    double higher = *m_tree.get<0>().nth(ct);
+                    qtl = (1 - target + ft) * lower + (1 - ct + target) * higher;
+                }
+                break;
+            case LOWER:
+                qtl = *m_tree.get<0>().nth(ft);
+                break;
+            case HIGHER:
+                qtl = *m_tree.get<0>().nth(ct);
+                break;
+            case MIDPOINT:
+                if (ft == target)
+                {
+                    qtl = *m_tree.get<0>().nth(ft);
+                }
+                else
+                {
+                    double lower = *m_tree.get<0>().nth(ft);
+                    double higher = *m_tree.get<0>().nth(ct);
```

Comment:
No need to call `nth` twice here, just keep the iterator from the first call and advance it to get the `higher` value.